lms/djangoapps/completion_api/views.py

```diff
@@ -67,9 +69,16 @@ def get_progress_queryset(self):
         """
         Build a base queryset of relevant StudentProgress objects.
         """
-        objs = StudentProgress.objects.filter(user=self.get_user())
+        objs = StudentProgress.objects.filter(user=self.user)
         return objs
 
+    def create_progress(self, course_id):
+        """
+        Build an empty StudentProgress object for the current user and given course.
+        """
+        obj = StudentProgress(user=self.user, course_id=course_id)
+        return obj
+
```

Comment:
Since I'd still have to pass in `self.user` and `course_id`, I can't see any reason why creating a separate method on the model would be helpful here?

Also, it means a separate PR on [progress-edx-platform-extensions](https://github.com/edx-solutions/progress-edx-platform-extensions/blob/1980336ef217c7c076fc12f8768962a2396c3dee/progress/models.py#L16), and honestly, I can't be stuffed 😄

Reply:
Legit.